core/src/main/scala/org/apache/spark/storage/BlockManagerDecommissioner.scala

```diff
@@ -207,7 +207,7 @@ private[storage] class BlockManagerDecommissioner(
       logInfo("Attempting to migrate all RDD blocks")
       while (!stopped && !stoppedRDD) {
         // Validate if we have peers to migrate to. Otherwise, give up migration.
-        if (bm.getPeers(false).isEmpty) {
+        if (!bm.getPeers(false).exists(x => x != FallbackStorage.FALLBACK_BLOCK_MANAGER_ID)) {
```

Comment:
Yes, it's a possible solution if we don't want to implement a migration to S3.

nit.
```scala
if (!bm.getPeers(false).exists(x => x != FallbackStorage.FALLBACK_BLOCK_MANAGER_ID)) {
if (!bm.getPeers(false).exists(_ != FallbackStorage.FALLBACK_BLOCK_MANAGER_ID)) {
```

Reply:
Will open a separate PR to address the migration to S3, but the main focus for this PR is to fix bugs that will cause decommissioned executors stay alive forever, and create a deadlock in some cases. This causes a deadlock in the following situation: 
1. All executors are marked as decommissioned, and we have a cap on number of executors
2. The migration of RDD blocks will never finish because of fallback storage
3. New tasks are launched and there are no executors eligible to run them due to decommission and no new executors can be launched because of the max is reached.